### PR TITLE
rgw: using differnt parameter names in ctor RGWInitDataSyncStatusCoroutine()

### PR DESCRIPTION
**Pull request by shashalu**
…ncStatusCoroutine

Signed-off-by: Shasha Lu <lu.shasha@eisoo.com>

### CODE REVIEW

**Comment by joscollin** [comment]
Jenkins retest this please.

**Comment by joscollin** [comment]
Jenkins retest this please.

**Comment by tchaikov** [comment]
retest this please. see #15406

**Comment by joscollin** [comment]
Jenkins retest this please.

**Comment by theanalyst** [comment]
This would not be needed as such, see https://stackoverflow.com/questions/6185020/initializing-member-variables-using-the-same-name-for-constructor-arguments-as-f/6185043#6185043 (thanks @rzarzynski  for bringing it to notice)

**Comment by joscollin** [comment]
@theanalyst Yes you are right, it is perfectly fine to use the same names for parameters. I knew this before reviewing. But the changed convention ( _var ) is better for easy reading, searching etc.

Similarly, there are lot of places the variable names are declared as i, j, k etc, which makes it difficult to read and search for the occurrence of those variables, unless we have an editor that highlights the occurrences.

So this is a fancy change, but sometimes it is useful.

File: src/rgw/rgw_data_sync.cc [hunk 1]
```diff
@@ -469,11 +469,11 @@ class RGWInitDataSyncStatusCoroutine : public RGWCoroutine {
...
   map<int, RGWDataChangesLogInfo> shards_info;
 public:
-  RGWInitDataSyncStatusCoroutine(RGWDataSyncEnv *_sync_env, uint32_t num_shards,
-                                 rgw_data_sync_status *status)
+  RGWInitDataSyncStatusCoroutine(RGWDataSyncEnv *_sync_env, uint32_t _num_shards,
+                                 rgw_data_sync_status *_status)
     : RGWCoroutine(_sync_env->cct), sync_env(_sync_env), store(sync_env->store),
       pool(store->get_zone_params().log_pool),
-      num_shards(num_shards), status(status) {
+      num_shards(_num_shards), status(_status) {
```
Comment (tchaikov) [inline src/rgw/rgw_data_sync.cc]: i don't think this change is helpful.

**Comment by tchaikov** [comment]
@joscollin the initialization list is always easy to read. using this "convention" does not improve the readability IMHO. and this is different from the naming like "j", "k".

File: src/rgw/rgw_data_sync.cc (hunk 1 above)
Comment (rzarzynski) [inline src/rgw/rgw_data_sync.cc]: Personally I prefer to keep the names of ctor's parameters and corresponding members coherent. Adding any prefix mess with the Vim's searching and highlighting via `*`.

I'm also against the underscores, sorry.

**Comment by yehudasa** [comment]
I like using underscore, it makes the scoping clearer. otoh, I don't think we should go and do any swiping change, as there's no real need for it.

**Comment by cbodley** [comment]
closing this one; there doesn't seem to be a strong preference either way, and I don't think it's worth the churn

**Pull request closed without merging**